src/main/java/com/ioet/bpm/people/services/PersonService.java

```diff
@@ -44,4 +49,16 @@ public boolean authenticationIdentityExists(String authenticationIdentity) {
     public boolean emailChanged(Person oldPerson, Person personToUpdate) {
         return !oldPerson.getAuthenticationIdentity().equals(personToUpdate.getAuthenticationIdentity());
     }
+
+    public Iterable<Person> findPeopleByEmail(String email) {
+        Iterable<Person> persons;
+        if (email == null) {
+            persons = this.personRepository.findAll();
+        } else {
+            Optional<Person> personOptional = personRepository.findPersonByAuthenticationIdentity(email);
+            persons = new ArrayList<>();
+            personOptional.ifPresent(((ArrayList<Person>) persons)::add);
```

Comment:
This is cleaner:
`Optional<Person> personOptional = personRepository.findPersonByAuthenticationIdentity(email);
            persons = newArrayList(personOptional.orElseGet(null)); `

Please test it :)

Reply:
i don't know, i don't want to have anything in the list when the person isn't found, not even null. if i have a null in there the function in PersonController doesn't recognize that the list is empty any more